### PR TITLE
Os cnn

### LINKED ISSUE #17: Omni Scale 1D CNN implementation
To convert OS 1D CNN into a scikit-learn compatible neural network library that's wraped with PyTorch.

### PR DESCRIPTION
**Pull request by aabduvakhobov**
#17

### CODE REVIEW

File: notebooks/os_cnn_classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
+      "Input \u001b[0;32mIn [4]\u001b[0m, in \u001b[0;36m<cell line: 3>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m accuracy_list\u001b[38;5;241m=\u001b[39m[]\n\u001b[1;32m      2\u001b[0m correct \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m\n\u001b[0;32m----> 3\u001b[0m N_test\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mlen\u001b[39m(\u001b[43mtest_dataset\u001b[49m)\n\u001b[1;32m      4\u001b[0m yhat \u001b[38;5;241m=\u001b[39m model\u001b[38;5;241m.\u001b[39mpredict(X_test)\n\u001b[1;32m      5\u001b[0m correct \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m (torch\u001b[38;5;241m.\u001b[39mtensor(yhat) \u001b[38;5;241m==\u001b[39m y_test)\u001b[38;5;241m.\u001b[39msum()\u001b[38;5;241m.\u001b[39mitem()\n",
+      "\u001b[0;31mNameError\u001b[0m: name 'test_dataset' is not defined"
+     ]
+    }
+   ],
+   "source": [
+    "accuracy_list=[]\n",
+    "correct = 0\n",
+    "N_test=len(test_dataset)\n",
```
Comment (dhavalsalwala) [inline notebooks/os_cnn_classification.ipynb]: Shouldn't this code len(test_dataset) be X_test.shape[0] ?

**Review by dhavalsalwala — CHANGES_REQUESTED** [review]
Hi @aabduvakhobov ,
Thanks for submitting the pull request. Just need one more change.

1) Delete the dataset folder - notebooks/OS_CNN. Share it via drive link in the comments.

File: notebooks/os_cnn_classification.ipynb (hunk 1 above)
Comment (aabduvakhobov) [reply notebooks/os_cnn_classification.ipynb]: Yeah you are right! I've corrected that.

**Review by dhavalsalwala — CHANGES_REQUESTED** [review]
Hi @aabduvakhobov ,
You still need to remove this directory: notebooks/OS_CNN. I see that you have added the dataset link in the comments. Thanks.

**Comment by aabduvakhobov** [comment]
hi Dhaval, my bad, just forgot it. Now got it removed :)

**Review by dhavalsalwala — CHANGES_REQUESTED** [review]
Hi @aabduvakhobov ,

Your unit test is failing. Please check here: https://github.com/IBM/sail/runs/6639291463?check_suite_focus=true

The possible solution is not to use a real-world dataset in unit tests. Your unit test cannot locate the path you have mentioned. Also, a unit test is more about your code being correct and less about the accuracy of the model. So, it is better to use a single epoch rather than going with 20 (20 takes a significant time on a Test VM).

You can create/re-use the synthetic dataset used here: https://github.com/IBM/sail/blob/main/tests/models/torch/conftest.py

**Comment by aabduvakhobov** [comment]
> Hi @aabduvakhobov ,
> 
> Your unit test is failing. Please check here: https://github.com/IBM/sail/runs/6639291463?check_suite_focus=true
> 
> The possible solution is not to use a real-world dataset in unit tests. Your unit test cannot locate the path you have mentioned. Also, a unit test is more about your code being correct and less about the accuracy of the model. So, it is better to use a single epoch rather than going with 20 (20 takes a significant time on a Test VM).
> 
> You can create/re-use the synthetic dataset used here: https://github.com/IBM/sail/blob/main/tests/models/torch/conftest.py

**Comment by aabduvakhobov** [comment]
Hi @dhavalsalwala ,
Thanks for the comment. Now I've added a synthetic dataset only to the unit test, while users of the notebook will have to download it from the google drive. Also number of epochs in the unit test was decreased to the minimum.

**Review by dhavalsalwala — APPROVED** [review]
The changes look ok to me.
